theoriq/api/v1alpha2/configure.py

```diff
@@ -19,3 +25,60 @@ def virtual_address(self) -> AgentAddress:
 
 
 ConfigureFn = Callable[[ConfigureContext, Any], None]
+"""A callable type for the configuration function.
+
+The function accepts:
+- A `ConfigureContext` instance to provide the configuration context.
+- Additional keyword arguments to build the configuration.
+"""
+
+IsLongRunningFn = Callable[[ConfigureContext, Any], bool]
+"""A callable type for checking if a configuration is long-running.
+
+:param
+- A `ConfigureContext` instance to provide the configuration context.
+- Additional keyword arguments to build the configuration.
+
+:returns
+- A boolean indicating whether the configuration is long-running.
+"""
+
+
+class AgentConfigurator:
+    """
+    A class to manage and execute configuration for agents.
+    """
+
+    def __init__(self, configure_fn: ConfigureFn, is_long_running_fn: IsLongRunningFn) -> None:
+        self.configure_fn = configure_fn
+        self.is_long_running_fn = is_long_running_fn
+
+    def __call__(self, configure_context: ConfigureContext, payload: Any, biscuit: TheoriqBiscuit, agent: Agent):
+        client = configure_context._protocol_client
+
+        try:
+            self.configure_fn(configure_context, payload)
+        except Exception as e:
+            logger.error(f"Failed to configure agent: {e}")
+            client.post_request_failure(biscuit, str(e), agent)
+        else:
```

Comment:
Add a `post_request_failure` and `post_request_success` on the `ConfigureContext` class. That class already contains the `agent` and an instance of the `protocol_client`. This will avoid to access a private field `client = configure_context._protocol_client` and simplify signatures